packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -147,6 +147,9 @@ Future<void> buildMacOS({
   if (configuration == null) {
     throwToolExit('Unable to find expected configuration in Xcode project.');
   }
+  // Specifying the architecture makes the destination clear to Xcode
+  final String arch = globals.platform.version.contains('arm64') ? 'arm64' : globals.platform.version.contains('x86_64') ? 'x86_64' : '';
```

Comment:
Sorry for the delay on this review.
Instead of string parsing, you should check `OperatingSystemUtils.hostPlatform`:
```suggestion
  final String arch = globals.os.hostPlatform == HostPlatform.darwin_arm64 ? 'arm64' : globals.platform.version.contains('x86_64') ? 'x86_64' : '';
```

Reply:
The empty string fallback is just in case it's not fetched, it does nothing and keeps the behaviour before this pr.

Reply:
Using an empty string also allows the old tests to pass normally. If I switch to using globals.os.hostPlatform, I might need to modify the existing tests, as the destination parameter is hard-coded in the current tests. I'm not sure if this is an acceptable practice.